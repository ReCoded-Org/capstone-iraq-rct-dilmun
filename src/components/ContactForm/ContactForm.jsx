import React from 'react'
import { useTranslation } from 'react-i18next'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

export default function ContactForm() {
  const { t } = useTranslation()
  const pnum = '+234081-1236-4568'
  const em = 'contact@Dilmun.com'

  return (
    <div className=" bg-white">
      <div className="relative flex items-top justify-center sm:items-center sm:pt-0  ">
        <div className="max-w-5xl  sm:px-6 lg:px-8 m-8 border-2 border-darkgray rounded-3xl">
          <h1 className="text-3xl sm:text-4xl text-blue mt-6 font-extrabold text-center mx-14 border-b-2">
            {t('cform.cont')}
          </h1>
          <div className="grid  grid-cols-1 md:grid-cols-2 mx-6 ">
            <form className=" flex flex-col  mt-2">
              <p className="text-normal text-lg sm:text-2xl mt-8">
                {t('cform.leve')}{' '}
              </p>
              <div>
                <label htmlFor="name" className="hidden">
                  {t('cform.name')}
                </label>
                <input
                  type="text"
                  name="name"
                  id="name"
                  placeholder={t('cform.name')}
                  className="w-full mt-2 py-3 px-1  bg-white  border  font-semibold focus:border-darkgray focus:outline-none"
                />
              </div>
              <div>
                <label htmlFor="email" className="hidden">
                  {t('cform.email')}
                </label>
                <input
                  type="email"
                  name="email"
                  id="email"
                  placeholder={t('cform.email')}
                  className="w-full mt-2 py-3 px-1 bg-white  border   font-semibold focus:border-darkgray focus:outline-none"
                />
              </div>
              <div>
                <label htmlFor="message" className="hidden">
                  {t('cform.message')}
                </label>
                <textarea
                  name="message"
                  id="message"
                  placeholder={t('cform.message')}
                  className="w-full mt-2 h-40 px-1 bg-white  border  font-semibold focus:border-darkgray focus:outline-none"
                />
              </div>
              <div>
                <button
                  type="submit"
                  className="w-full mb-10
                                                     bg-blue hover:bg-blue-dark text-white font-bold py-3 px-24  mt-3 hover:bg-darkBlue transition ease-in-out duration-300"
                >
                  {t('cform.send')}
                </button>
              </div>
            </form>
            <div className="flex items-center mt-4 hidden md:flex ">
              <div className="flex-shrink-0 md:mx-20 mx-auto justify-center md:text-left md:justify-start ">
                <div>
                  <a href={`tel:${pnum}`}>
                    <FontAwesomeIcon
                      icon="phone"
                      className="hover:text-green "
                    />
                    <span>+234081-1236-4568</span>
                  </a>
                </div>
                <div>
                  <a href={`mailto:${em}`}>
                    <FontAwesomeIcon
                      icon="envelope"
                      className="hover:text-green "
                    />{' '}
                    <span>contact@Dilmun.com</span>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className=" justify-center text-center max-w-screen-md p-8 lg:mx-auto">
        <p className="text-lg  sm:text-2xl text-blue font-bold ">
          {t('cform.disc')}
        </p>
        <p className="text-base sm:text-lg text-black ">{t('cform.desc')}</p>
      </div>
    </div>
  )
}
