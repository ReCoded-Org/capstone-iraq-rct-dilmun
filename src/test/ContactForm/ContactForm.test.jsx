import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter as  Router} from 'react-router-dom'
import ContactForm from "../../components/ContactForm/ContactForm";

it ("Contact Form renders correctly", () => {
  const tree = renderer.create(<Router> 
    <ContactForm/>
     </Router>).toJSON();
  expect(tree).toMatchSnapshot();
});