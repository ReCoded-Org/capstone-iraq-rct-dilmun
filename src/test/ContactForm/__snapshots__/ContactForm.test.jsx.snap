// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Form renders correctly 1`] = `
Array [
  " ",
  <div
    className=" bg-white"
  >
    <div
      className="relative flex items-top justify-center sm:items-center sm:pt-0  "
    >
      <div
        className="max-w-5xl  sm:px-6 lg:px-8 m-8 border-2 border-darkgray rounded-3xl"
      >
        <h1
          className="text-3xl sm:text-4xl text-blue mt-6 font-extrabold text-center mx-14 border-b-2"
        >
          cform.cont
        </h1>
        <div
          className="grid  grid-cols-1 md:grid-cols-2 mx-6 "
        >
          <form
            className=" flex flex-col  mt-2"
          >
            <p
              className="text-normal text-lg sm:text-2xl mt-8"
            >
              cform.leve
               
            </p>
            <div>
              <label
                className="hidden"
                htmlFor="name"
              >
                cform.name
              </label>
              <input
                className="w-full mt-2 py-3 px-1  bg-white  border  font-semibold focus:border-darkgray focus:outline-none"
                id="name"
                name="name"
                placeholder="cform.name"
                type="text"
              />
            </div>
            <div>
              <label
                className="hidden"
                htmlFor="email"
              >
                cform.email
              </label>
              <input
                className="w-full mt-2 py-3 px-1 bg-white  border   font-semibold focus:border-darkgray focus:outline-none"
                id="email"
                name="email"
                placeholder="cform.email"
                type="email"
              />
            </div>
            <div>
              <label
                className="hidden"
                htmlFor="message"
              >
                cform.message
              </label>
              <textarea
                className="w-full mt-2 h-40 px-1 bg-white  border  font-semibold focus:border-darkgray focus:outline-none"
                id="message"
                name="message"
                placeholder="cform.message"
              />
            </div>
            <div>
              <button
                className="w-full mb-10
 bg-blue hover:bg-blue-dark text-white font-bold py-3 px-24  mt-3 hover:bg-darkBlue transition ease-in-out duration-300"
                type="submit"
              >
                cform.send
              </button>
            </div>
          </form>
          <div
            className="flex items-center mt-4 hidden md:flex "
          >
            <div
              className="flex-shrink-0 md:mx-20 mx-auto justify-center md:text-left md:justify-start "
            >
              <div>
                <a
                  href="tel:+234081-1236-4568"
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-phone fa-w-16 hover:text-green "
                    data-icon="phone"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                  <span>
                    +234 081-1236-4568
                  </span>
                </a>
              </div>
              <div>
                <a
                  href="mailto:contact@Dilmun.com"
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-envelope fa-w-16 hover:text-green "
                    data-icon="envelope"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                    
                  <span>
                    contact@Dilmun.com
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className=" justify-center text-center max-w-screen-md p-8 lg:mx-auto"
    >
      <p
        className="text-lg  sm:text-2xl text-blue font-bold "
      >
        cform.disc
      </p>
      <p
        className="text-base sm:text-lg text-black "
      >
        cform.desc
      </p>
    </div>
  </div>,
  " ",
]
`;
